### ISSUE TITLE
Usage of default buffer for SYSCOLUMNS_FULL

### ISSUE THREAD

**Issue by dominickdm** [Riverside-Software/sonar-openedge#530]
Line "arguments =CAST..." in the testclass below is resulting in a 'BufferUsage' rule violation: usage of default buffer for SYSCOLUMNS_FULL.

```
USING Progress.Lang.*.
USING System.Array FROM ASSEMBLY.
USING Progress.Util.TypeHelper FROM ASSEMBLY.

BLOCK-LEVEL ON ERROR UNDO, THROW.


CLASS testclass
    CONSTRUCTOR PUBLIC testclass (  ):

        DEFINE VARIABLE Arguments AS CLASS "System.Object[]":U NO-UNDO.
        Arguments = CAST(Array:CreateInstance(TypeHelper:GetType("System.Object":U), 1), "System.Object[]":U).

    END CONSTRUCTOR.

END CLASS.
```

**Comment by gquerret** [comment]
Issue fixed in ANTLR4 parser (released later), but temp fix will be available in 2.3.0 (no issue reported on SYS* tables)
